File: nest-todo/src/todo/todo.controller.ts
import { Controller, Post, Body, Get, Put, Delete, Param } from '@nestjs/common';
import { TodoService } from './todo.service';
import { Todo } from './todo.entity';

@Controller('todo')
export class TodoController {

  constructor(private service: TodoService) { }

  @Post()
  async create(@Body('description') description: string) {
    return await this.service.save(description);
  }

  @Get()
  async listAllTodos() {
    return await this.service.findAll();
  }

  @Put()
  async updateTodo(@Body() todo: Todo) {
    return await this.service.update(todo);
  }

  @Delete(':id')
  async removeTodo(@Param('id') id: number) {
    return await this.service.remove(id);
  }

}

File: flutter_todo(Mobile)/lib/todo/todo.controller.dart
import 'package:flutter_todo/todo/todo.model.dart';
import 'package:flutter_todo/todo/todo.service.dart';
import 'package:mobx/mobx.dart';

part 'todo.controller.g.dart';

class TodoController = _TodoControllerBase with _$TodoController;

abstract class _TodoControllerBase with Store {
  @observable
  List<Todo> todos = [];

  @action
  fetchTodos() async {
    todos = await TodoService().listAll();
  }

  @action
  create(String description) async {
    await TodoService().create(description);
    fetchTodos();
  }

  @action
  remove(int id) async {
    await TodoService().remove(id);
    fetchTodos();
  }

  @action
  update(Todo todo) async {
    await TodoService().update(todo);
    fetchTodos();
  }


}
